### PR TITLE
Add smooth movement for player effects

### PR DESCRIPTION
**Pull request by kaspermelheim**
Player now moves smoothly around the board. Looks more clean. Rotation effect still needs to be added.

### CODE REVIEW

File: src/main/java/inf112/isolasjonsteamet/roborally/actions/ActionProcessor.java [hunk 1]
```diff
@@ -13,4 +13,6 @@
 	 * @param player The player to run the action for.
 	 */
 	void performActionNow(Player player, Action action);
+
+	void scheduleAction(Player player, Action action);
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/actions/ActionProcessor.java:20]: Javadoc

File: src/main/java/inf112/isolasjonsteamet/roborally/actions/Move.java [hunk 2]
```diff
@@ -14,12 +16,16 @@
...
 
+	private PlayerEffect playerEffect;
+
 	public Move(Orientation direction, int numMoves) {
 		this.direction = direction;
 		this.numMoves = numMoves;
 	}
 
-	@Override
+	/**
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/actions/Move.java:26]: Add back `@Override` and use `{@inheritDoc}`

File: src/main/java/inf112/isolasjonsteamet/roborally/app/RoboRallyGame.java [hunk 3]
```diff
@@ -93,22 +95,10 @@ public void create() {
 		table.add(new MapRendererWidget(board, 100));
 		table.row();
 
-		var bottomConsole = new PrintStreamLabel(2, System.out, skin, "default-font", Color.WHITE);
+		var bottomConsole = new PrintStreamLabel(3, System.out, skin, "default-font", Color.WHITE);
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/app/RoboRallyGame.java]: Merge conflict gone wrong here. You have removed tons of work from master.

File: src/main/java/inf112/isolasjonsteamet/roborally/app/RoboRallyGame.java [hunk 4]
```diff
@@ -227,12 +223,19 @@ public void render() {
 	/**
 	 * {@inheritDoc}
 	 */
+	@Override
 	public void performActionNow(Player player, Action action) {
-		final Coordinate oldPos = activePlayer.getPos();
+		@SuppressWarnings("VariableDeclarationUsageDistance")
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/app/RoboRallyGame.java]: If you need to ignore this warning, declare the field as final instead.

File: src/main/java/inf112/isolasjonsteamet/roborally/board/ClientBoard.java [hunk 5]
```diff
@@ -16,4 +19,14 @@
 	 * How large the tiles in the texture atlas are in pixels.
 	 */
 	int getTextureTileSize();
+
+	void show(Player player);
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/board/ClientBoard.java:26]: Javadoc

File: src/main/java/inf112/isolasjonsteamet/roborally/effects/PlayerEffect.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+	private static final TextureRegion PLAYER_TEXTURE;
+
+	static {
+		Texture playerTx = new Texture("player.png");
+		final TextureRegion[][] tReg = TextureRegion.split(playerTx, 300, 300);
+		PLAYER_TEXTURE = tReg[0][0];
+	}
+
+	Orientation dir;
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/effects/PlayerEffect.java]: Make this private

File: src/main/java/inf112/isolasjonsteamet/roborally/gui/MapRendererWidget.java [hunk 7]
```diff
@@ -31,16 +30,19 @@ public MapRendererWidget(ClientBoard board, int tileSize) {
...
+	/**
+	 * Draw method.
+	 */
 	@Override
 	public void draw(Batch batch, float parentAlpha) {
 		super.draw(batch, parentAlpha);
 		// The map renderer does not behave well together with scene2d's sprite batch,
 		// so we need to stop the current one, and restart it after we have rendered the map.
 		batch.end();
-
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/gui/MapRendererWidget.java]: Don't remove the newlines here. They're useful to indicate which batch is active where

File: src/test/java/inf112/isolasjonsteamet/roborally/PlayerSpec.java [hunk 8]
```diff
@@ -91,6 +92,7 @@ private void assertPlayerPos(Player player, Coordinate coord) {
 
 	@Override
 	public void performActionNow(Player player, Action action) {
+	    action.initialize(board, player);
```
Comment (Katrix) [inline src/test/java/inf112/isolasjonsteamet/roborally/PlayerSpec.java]: Formatting

File: src/test/java/inf112/isolasjonsteamet/roborally/PlayerSpec.java [hunk 9]
```diff
@@ -1,27 +1,28 @@
...
 import inf112.isolasjonsteamet.roborally.players.PlayerImpl;
 import inf112.isolasjonsteamet.roborally.tiles.TileType;
 import inf112.isolasjonsteamet.roborally.tiles.Tiles;
 import inf112.isolasjonsteamet.roborally.util.Coordinate;
 import inf112.isolasjonsteamet.roborally.util.Orientation;
-import java.util.List;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import java.util.List;
```
Comment (Katrix) [inline src/test/java/inf112/isolasjonsteamet/roborally/PlayerSpec.java]: Don't move the imports

**Review by Katrix — CHANGES_REQUESTED** [review]

File: src/main/java/inf112/isolasjonsteamet/roborally/gui/MapRendererWidget.java [hunk 10]
```diff
@@ -31,6 +30,9 @@ public MapRendererWidget(ClientBoard board, int tileSize) {
 		sizeChanged();
 	}
 
+	/**
```
Comment (Katrix) [inline src/main/java/inf112/isolasjonsteamet/roborally/gui/MapRendererWidget.java]: `{@inheritDoc}` in here. Even more important for some of the other methods in here, as the JavaDoc specified here is just plain wrong in some cases.

**Review by Katrix — APPROVED** [review]

**Review by espensva — APPROVED** [review]

**Pull request merged**